src/client.rs

```diff
@@ -240,6 +241,63 @@ impl HttpClientBuilder {
         self.configure(map)
     }
 
+    /// Set a default header to be passed with every request
+    ///
+    /// NOTE: In case there is an error in parsing the HeaderName or HeaderValue
+    /// the tuple is silently discarded.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use isahc::prelude::*;
+    /// #
+    /// let client = HttpClient::builder()
+    ///     .default_header("some-header", "some-value")
+    ///     .build()?;
+    /// # Ok::<(), Box<dyn std::error::Error>>(())
+    /// ```
+    pub fn default_header(
+        mut self,
+        key: impl TryInto<HeaderName>,
+        value: impl TryInto<HeaderValue>,
+    ) -> Self {
+        match key.try_into() {
+            Ok(key) => match value.try_into() {
+                Ok(value) => {
+                    if let Some(headers_map) = self.defaults.get_mut::<DefaultHeaderMap>() {
+                        headers_map.0.append(key, value);
+                    } else {
+                        let mut header = http::HeaderMap::with_capacity(1);
+                        header.insert(key, value);
+                        self.defaults.insert(DefaultHeaderMap(header));
+                    }
+                    self
+                }
+                Err(_e) => self,
+            },
+            Err(_e) => self,
+        }
+    }
+
+    /// Get the underlying HeaderMap from the current client-builder.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use isahc::prelude::*;
+    /// #
+    /// let mut builder = HttpClient::builder()
+    ///     .default_header("some-header", "some-value");
+    /// let header_opt = builder.default_headers_mut();
+    /// # Ok::<(), Box<dyn std::error::Error>>(())
+    /// ```
+    pub fn default_headers_mut(&mut self) -> Option<&mut http::header::HeaderMap<HeaderValue>> {
```

Comment:
Returning an `Option` here doesn't strike me as being very easy for consumers to use. I would just add a new field to the builder (`default_headers: http::HeaderMap<http::HeaderValue>`) so that we always have something to return here, even if no one uses it.